nixpkgs.nix

```diff
@@ -0,0 +1,2 @@
+# Import local nixpkg.json which pins all our Nix packages
+import (builtins.fetchTarball (builtins.fromJSON (builtins.readFile ./nixpkgs.json)))
```

Comment:
Might want to consider using a tool like npins or niv here, it will also make updating the pin much more pleasant

Reply:
I was looking into niv, but I could not see how/if it is used for PrivateStorageio!
(I've just made a copy/paste :-)

Reply:
@exarkun : how do we update nixpkgs in PrivateStorageio?

Reply:
The shell script `ci-tools/update-nixpkgs` calls, in its function `update_nixpkgs` on [line 160](https://whetstone.private.storage/privatestorage/PrivateStorageio/-/blob/develop/ci-tools/update-nixpkgs#L160), `update-nixpkgs` defined in `shell.nix`. That in turn calls the python script `tools/update-nixpkgs` to update `nixpkgs.json`.

The python script already broke once because the HTTP library it uses changed its API and we didn't have its version pinned.

That's why the moment @exarkun turns his back I'll replace the whole thing by my 12 line `update-nixpkgs.sh` shell script I have still sitting in my working directory which just creates a new JSON from a heredoc hr hr hr

Reply:
Thank you @hacklschorsch.
I was surprised there is not some nix command to do this...
And I was already trying to implement this in bash. :-)
I guess I'll not go further here.

Reply:
FWIW I think there are nix commands built-in when using flakes …

Reply:
I did not dare pushing this experiment to the flake level :-)
Nice to know this could help to simplify the update of the nixpkgs.

I'll see later if I can spend some extra time to improve this (with `npins`, `niv` or `flake`).
Maybe interesting case for a "Nix Intro" session?

Reply:
I won't be there this week because national holiday, but next week I could talk a bit about these

Reply:
No business value in adding yet another way to update one three line json file.